dso-l2/src/main/java/com/tc/objectserver/impl/TopologyManager.java

```diff
@@ -0,0 +1,129 @@
+package com.tc.objectserver.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.tc.management.AbstractTerracottaMBean;
+import com.tc.management.TerracottaManagement;
+
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+
+import static java.lang.String.join;
+
+public class TopologyManager {
+
+  private Topology topology;
+  private volatile TopologyMbean topologyMbean;
+  private final List<TopologyListener> listeners = new ArrayList<>();
+
+  private TopologyManager() {
+  }
+
+  private static final TopologyManager INSTANCE = new TopologyManager();
+
+  public static TopologyManager get() {
+    return INSTANCE;
```

Comment:
Can this be done without singletons?

Reply:
Yes but TopologyManager needs to be passed to all components where it is used. will try that.